From 6d3a6909b1cf7b6423a762d52b182fb01a279c2b Mon Sep 17 00:00:00 2001
From: Max Roncace <me@caseif.net>
Date: Sun, 27 Dec 2015 18:00:23 -0500
Subject: [PATCH] Disable class name validation


diff --git a/src/org/jetbrains/java/decompiler/main/ClassWriter.java b/src/org/jetbrains/java/decompiler/main/ClassWriter.java
index b90f2ec..44f3dbb 100644
--- a/src/org/jetbrains/java/decompiler/main/ClassWriter.java
+++ b/src/org/jetbrains/java/decompiler/main/ClassWriter.java
@@ -673,7 +673,7 @@ public class ClassWriter {
           buffer.append(' ');
         }
 
-        buffer.append(toValidJavaIdentifier(name));
+        buffer.append(name); // Nocturne: removed name validation
         buffer.append('(');
 
         // parameters
-- 
2.5.3.windows.1

