From 16dacc4b01688d1bc630cc6b2c3fed51f7e9b724 Mon Sep 17 00:00:00 2001
From: Max Roncace <me@caseif.net>
Date: Thu, 21 Jan 2016 13:19:07 -0500
Subject: [PATCH] Disable inner class name simplification for processed classes


diff --git a/src/org/jetbrains/java/decompiler/main/ClassesProcessor.java b/src/org/jetbrains/java/decompiler/main/ClassesProcessor.java
index be5af07..f304142 100644
--- a/src/org/jetbrains/java/decompiler/main/ClassesProcessor.java
+++ b/src/org/jetbrains/java/decompiler/main/ClassesProcessor.java
@@ -214,7 +214,9 @@ public class ClassesProcessor {
                 //}
 
                 nestedNode.type = (Integer)arr[2];
-                nestedNode.simpleName = (String)arr[1];
+                if (!nestedNode.simpleName.startsWith(ConstantsUtil.Nocturne.CLASS_PREFIX)) {
+                  nestedNode.simpleName = (String) arr[1];
+                }
                 nestedNode.access = (Integer)arr[3];
 
                 if (nestedNode.type == ClassNode.CLASS_ANONYMOUS) {
-- 
2.5.3.windows.1

