From 65c434d90088cd1146d1363a26c8dc17ffdd8573 Mon Sep 17 00:00:00 2001
From: Jamie Mansfield <dev@jamierocks.uk>
Date: Thu, 15 Nov 2018 18:04:12 +0000
Subject: [PATCH] Handle generics with nocturne processing


diff --git a/src/org/jetbrains/java/decompiler/modules/decompiler/ExprProcessor.java b/src/org/jetbrains/java/decompiler/modules/decompiler/ExprProcessor.java
index 537c470..ffc649b 100644
--- a/src/org/jetbrains/java/decompiler/modules/decompiler/ExprProcessor.java
+++ b/src/org/jetbrains/java/decompiler/modules/decompiler/ExprProcessor.java
@@ -748,6 +748,9 @@ public class ExprProcessor implements CodeConstants {
         // FIXME: a warning should be logged
         ret = UNDEFINED_TYPE_STRING;
       }
+      if (!ret.startsWith(ConstantsUtil.Nocturne.CLASS_PREFIX)) {
+        return ConstantsUtil.Nocturne.CLASS_PREFIX + ret + ConstantsUtil.Nocturne.CLASS_SUFFIX;
+      }
       return ret;
     }
 
diff --git a/src/org/jetbrains/java/decompiler/struct/gen/generics/GenericType.java b/src/org/jetbrains/java/decompiler/struct/gen/generics/GenericType.java
index 9b111e5..6845f4f 100644
--- a/src/org/jetbrains/java/decompiler/struct/gen/generics/GenericType.java
+++ b/src/org/jetbrains/java/decompiler/struct/gen/generics/GenericType.java
@@ -16,6 +16,7 @@
 package org.jetbrains.java.decompiler.struct.gen.generics;
 
 import org.jetbrains.java.decompiler.code.CodeConstants;
+import org.jetbrains.java.decompiler.code.ConstantsUtil;
 import org.jetbrains.java.decompiler.main.DecompilerContext;
 import org.jetbrains.java.decompiler.modules.decompiler.ExprProcessor;
 import org.jetbrains.java.decompiler.struct.gen.VarType;
@@ -285,7 +286,7 @@ public class GenericType extends VarType {
     else {
       clsName = DecompilerContext.getImportCollector().getShortName(parent.value.replace('/', '.')) + "." + value.substring(value.lastIndexOf('.') + 1);
     }
-    return clsName + getTypeArguments();
+    return ConstantsUtil.Nocturne.CLASS_PREFIX + clsName + ConstantsUtil.Nocturne.CLASS_SUFFIX + getTypeArguments();
   }
 
   private String getTypeArguments() {
-- 
2.19.1

